Nearmap 2.x

jevonkendon opened the pull request:
Rename the assembly and namespace to make it distinct from regular statsd.

- Update `NuGet.exe` to 2.8.60318.667 (was 2.0.30619.9000)
- Nearmap-ify the nupsec file

Commit:
BE-1450: Rename assembly and root namesapce to DogStatsdClient

Commit:
BE-1450: update .gitignore
- ignore Rider
- ignore packages

richarddbarnett commented:
why not merge this?

richarddbarnett commented:
delete files which you later .gitignored

richarddbarnett commented:
Document somewhere commands to build, test & nuget push this.

jevonkendon commented:
I thought a long lived branch might be easier to manage; that is, to compare and update, and make clear our changes. You think it should be merged?

Commit:
BE-1450: delete rider files

richarddbarnett commented:
I doubt there will be any changes made to the upstream 2.x branch while we're using this repo, or (most likely) ever; so we won't have to merge anything from upstream. Our changes should be clear from the commit?

richarddbarnett commented:
Branch name could have been better, but too late to change it now.

Commit:
BE-1450: configure nuget package for nearmap
- update the metadata for the assemblies and nuget package to use the nearmap semantic versioning strategy
- update NuGet.exe to our current version used in our microservices

jevonkendon commented:
The branch name fitted with my original idea of a long lived branch. I can re-branch once we are happy with this PR.

Commit:
BE-1450: revert assembly info files

Commit:
BE-1450: update nuspec

Commit:
BE-1450: update README to include build, test and nuget publish

Commit:
BE-1450: tweak

jevonkendon commented:
I had initially played with our build scripts to achieve an early package, and ended up with the `AssemblyInfo` files being updated with our versioning. Though it seemed harmless enough, `xbuild` refused to process it. I reverted those changes.

Commit:
BE-1450: fix nuget pack command

richarddbarnett commented on README.md:
````diff
@@ -204,8 +197,21 @@ _* Mono JIT compiler (tested with version 4.6.0)_
+
+2. Publish the NuGet package
+  ```
+  curl -v --fail -s -S -X PUT -H "X-NuGet-ApiKey:<secretkey>" -F package=@DogStatsD-CSharp-Client.<version>.nupkg http://nuget.development.nearmap.com/api/packages
````
could probably use `NuGet.exe push` - our makefile doesn't only because I found it unreliable for the Framework, which pushes many packages consecutively - for single packages it was ok.

richarddbarnett commented on src/StatsdClient/StatsdClient.nuspec:
```diff
@@ -1,16 +1,20 @@
+    <iconUrl>https://si0.twimg.com/profile_images/1380708189/dd_logo2_icon.png</iconUrl>
+    <copyright>Copyright &#169; 2018</copyright>
+    <tags>statsd dogstatsd datadog nearmap</tags>
+  </metadata>
```
this should probably have different `id`, `version` since Klondike is a local cache for 3rd-party packages & so it probably already has the Datadog package cached.

jevonkendon replied:
Yeah, it's somewhat hidden at the moment.

jevonkendon replied:
I've decided to leave the `version` the same, so we can easily match it to DataDog's version (i.e we might need to know if it's `2.x` or `3.x`, if we upgrade .NET). I imagine at most we'll only ever have one Nearmap nuget package per DataDog nuget package.

richarddbarnett commented on src/StatsdClient/StatsdClient.nuspec:
```diff
@@ -1,16 +1,20 @@
+    <title>DogStatsD-CSharp-Client</title>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <licenseUrl>https://github.com/DataDog/dogstatsd-csharp-client/blob/master/MIT-LICENCE.md</licenseUrl>
+    <description>A DogStatsD client for C# (v2.2.1). Forked by Nearmap to adjust the assembly name and root namespace to avoid conflict with statsd. DogStatsD is an extension of the StatsD metric server for use with Datadog. For more information visit http://datadoghq.com.</description>
```
"...to avoid conflict with https://www.nuget.org/packages/StatsdClient/" maybe

richarddbarnett commented on src/StatsdClient/StatsdClient.nuspec:
```diff
@@ -1,16 +1,20 @@
+    <owners>Datadog,Nearmap</owners>
+    <title>DogStatsD-CSharp-Client</title>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <licenseUrl>https://github.com/DataDog/dogstatsd-csharp-client/blob/master/MIT-LICENCE.md</licenseUrl>
```
should this be updated too?

richarddbarnett commented on README.md:
````diff
@@ -204,8 +197,21 @@ _* Mono JIT compiler (tested with version 4.6.0)_
 
+1. Build the NuGet package
+  ```
+  mono lib/NuGet.exe pack src/StatsdClient/StatsdClient.nuspec -Version '<version>' -BasePath <path>/src/StatsdClient -IncludeReferencedProjects
````
Is `-Version` required when `<version>` is already in the `.nuspec`?

jevonkendon replied:
Oh yes. Much better.

richarddbarnett commented on README.md:
```diff
@@ -1,6 +1,11 @@
-# DogStatsD for C#
+# Nearmap #
+
+Fork of DogStatsD for C#, in order to solve an assembly clash with the original StatsdClient for C# (i.e. for regular statsd). This allows a C# project to reference both the DogStatsD and StatsdClient assemblies side-by-side.
```
Assembly and namespace clashes.

Commit:
BE-1450: rename nuget package and simplify nuget publish steps
